File: src/main/java/org/usfirst/frc/team6325/robot/commands/Auto/AutoChooser.java
package org.usfirst.frc.team6325.robot.commands.Auto;


public class AutoChooser 
{
	public enum AutoPosition 
	{
	    LEFT("Left"),
	    MIDDLE("Middle"),
	    RIGHT("Right");

	    private final String name;

	    AutoPosition(String name){ this.name = name; }

	    public String getName(){ return name; }
	}

	public enum AutoPreference 
	{
	    SCALE("Scale"),
	    SWITCH("Switch"),
		SIMPLE("Simple"),
		BASELINE("Baseline");

	    private final String name;

	    AutoPreference(String name){ this.name = name; }

	    public String getName(){ return name; }
	}
	
	public enum AutoCubes 
	{
		ONE("One"), 
		TWO("Two"), 
		THREE("Three"),
		ONEONE("OneOne"), 
		ONETWO("OneTwo"); 

	    
	    private final String name;

	    AutoCubes(String name){ this.name = name; }

	    public String getName(){ return name; }
	}
}

File: src/main/java/org/usfirst/frc/team6325/robot/commands/Auto/AutoChooser.kt
package org.usfirst.frc.team6325.robot.commands.Auto


public class AutoChooser
{
    public enum class AutoPosition
    {
        LEFT("Left"),
        MIDDLE("Middle"),
        RIGHT("Right")

        var name: String = null

        fun AutoPosition(name: String) = this.name = name

        public fun getName(): String = return name
    }

    public enum class AutoPreference
    {
        SCALE("Scale"),
        SWITCH("Switch"),
        SIMPLE("Simple"),
        BASELINE("Baseline")

        var name: String = null
        
        fun AutoPreference(name: String) = this.name = name

        public fun getName(): String = return name
    }

    public enum class AutoCubes
    {
	    ONE("One"), 
		TWO("Two"), 
		THREE("Three"),
		ONEONE("OneOne"), 
        ONETWO("OneTwo")

        var name: String = null

        fun AutoCubes(name: String) = this.name = name

        public fun getName(): String = return name
    }
}